tests/scripts/gather-code-coverage.sh

```diff
@@ -7,6 +7,17 @@
 
 # First parameter is the name of the generated info file
 INFO_FILE=${1:-coverage.info}
+BLUECHI_VERSION=${2:-unknown}
+
+# Find the bluechi debug directory dynamically to extract version
+BLUECHI_DEBUG_DIR=$(find /usr/src/debug -maxdepth 1 -type d -name 'bluechi-*' | head -1)
+if [ -z "$BLUECHI_DEBUG_DIR" ]; then
+    echo "Error: No bluechi-* directory found in /usr/src/debug"
+    exit 1
+fi
+
+# Construct the BUILD directory path using the extracted version
+BLUECHI_BUILD_DIR="/github/home/rpmbuild/BUILD/bluechi-${BLUECHI_VERSION}"
```

Comment:
I think we could use the `$HOME` env var:
```suggestion
BLUECHI_BUILD_DIR="${HOME}/rpmbuild/BUILD/bluechi-${BLUECHI_VERSION}"
```
Not really sure this would work on local execution of the tests, but at least would be a bit more flexible.

Reply:
${HOME} in the CI is root

Reply:
This is not supposed to be home /github/home is the dir that lcov ran in the other container

Reply:
In which other container? So where is this coverage coming from? Its been a longer time since I had a look there, so not sure.

Reply:
You run the lcov on the CI. But when you run the integration test, some containers are coming up to run bluechi controller or bluechi agent, and at the end of their life, they gather the gcda file to copy it to the main machine (CI). You can see it in the machine.py file that runs the gather-code-coverage.py. I hope it is clear now.